packages/advanced-logic/src/extensions/payment-network/erc20/address-based.ts

```diff
@@ -1,142 +1,48 @@
+export default class Erc20AddressBasedPaymentNetwork extends AddressBasedPaymentNetwork {
+  public constructor() {
+    super(ExtensionTypes.ID.PAYMENT_NETWORK_ERC20_ADDRESS_BASED, CURRENT_VERSION);
   }
 
-  return AddressBased.createAddPaymentAddressAction(
-    ExtensionTypes.ID.PAYMENT_NETWORK_ERC20_ADDRESS_BASED,
-    addPaymentAddressParameters,
-  );
-}
-
-/**
- * Creates the extensionsData to add a refund address
- *
- * @param addRefundAddressParameters extensions parameters to create
- *
- * @returns IAction the extensionsData to be stored in the request
- */
-function createAddRefundAddressAction(
-  addRefundAddressParameters: ExtensionTypes.PnAddressBased.IAddRefundAddressParameters,
-): ExtensionTypes.IAction {
-  if (
-    addRefundAddressParameters.refundAddress &&
-    !isValidAddress(addRefundAddressParameters.refundAddress)
-  ) {
-    throw Error('refundAddress is not a valid ethereum address');
+  /**
+   * Check if an ethereum address is valid
+   *
+   * @param {string} address address to check
+   * @returns {boolean} true if address is valid
+   */
+  protected isValidAddress(address: string): boolean {
+    return walletAddressValidator.validate(address, 'ethereum');
   }
 
-  return AddressBased.createAddRefundAddressAction(
-    ExtensionTypes.ID.PAYMENT_NETWORK_ERC20_ADDRESS_BASED,
-    addRefundAddressParameters,
-  );
-}
-
-/**
- * Applies the extension action to the request
- * Is called to interpret the extensions data when applying the transaction
- *
- * @param extensionsState previous state of the extensions
- * @param extensionAction action to apply
- * @param requestState request state read-only
- * @param actionSigner identity of the signer
- *
- * @returns state of the request updated
- */
-function applyActionToExtension(
-  extensionsState: RequestLogicTypes.IExtensionStates,
-  extensionAction: ExtensionTypes.IAction,
-  requestState: RequestLogicTypes.IRequest,
-  actionSigner: IdentityTypes.IIdentity,
-  timestamp: number,
-): RequestLogicTypes.IExtensionStates {
-  if (
-    requestState.currency.type !== RequestLogicTypes.CURRENCY.ERC20 ||
-    (requestState.currency.network && !supportedNetworks.includes(requestState.currency.network))
-  ) {
-    throw Error(
-      `This extension can be used only on ERC20 requests and on supported networks ${supportedNetworks.join(
-        ', ',
-      )}`,
-    );
+  protected validate(
+    request: RequestLogicTypes.IRequest,
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    _extensionAction: ExtensionTypes.IAction,
+  ): void {
+    if (
+      request.currency.type !== RequestLogicTypes.CURRENCY.ERC20 ||
+      (request.currency.network && !supportedNetworks.includes(request.currency.network))
+    ) {
+      throw Error(
+        `This extension can be used only on ERC20 requests and on supported networks ${supportedNetworks.join(
+          ', ',
+        )}`,
+      );
+    }
```

Comment:
it feels like `AddressBasedPaymentNetwork` should contain `supportedNetworks` and `supportedTypes` fields, and that the `validate` method could be removed from all child classes

Reply:
I was thinking about this but for [any-to-erc20-proxy.ts](packages/advanced-logic/src/extensions/payment-network/any-to-erc20-proxy.ts) we have a different way of validating the currency.